### PR TITLE
FileMock fixes

### PR DESCRIPTION
**Pull request by milo**

### CODE REVIEW

File: src/Framework/FileMock.php [hunk 1]
```diff
@@ -40,16 +48,51 @@ public static function create($content, $extension = NULL)
...
+		if (!self::$registered) {
+			stream_wrapper_register(self::PROTOCOL, __CLASS__);
+			self::$registered = TRUE;
+		}
+	}
+
+
 	public function stream_open($path, $mode)
 	{
+		if (!preg_match('#^([rwaxc]).*?(\+)?#', $mode, $m)) {
```
Comment (milo) [inline src/Framework/FileMock.php]: Not sure, how PCRE should be strict. I tried, that first letter must match mode, others can be mess.

**Comment by milo** [comment]
@dg What do you think about this approach? It is mainly fix for #291. Not sure it shoud be 1.7.x included.

File: src/Framework/FileMock.php [hunk 2]
```diff
@@ -40,16 +48,51 @@ public static function create($content, $extension = NULL)
 	}
 
 
+	public static function register()
```
Comment (milo) [inline src/Framework/FileMock.php]: Purpose: when someone want to use mock:// before `FileMock::create()`, e.g. testing with `mock://not-exist-yet`.

**Comment by milo** [comment]
Not sure why it fails in here (https://github.com/nette/tester/blob/84bcf2514baa8e4775cb27461de7bac3dcfae519/tests/CodeCoverage/CloverXMLGenerator.phpt#L27)

File: src/Framework/FileMock.php [hunk 3]
```diff
@@ -40,16 +48,51 @@ public static function create($content, $extension = NULL)
...
+	{
+		if (!self::$registered) {
```
Comment (dg) [inline src/Framework/FileMock.php]: Maybe can be replaced with stream_get_wrappers() https://github.com/nette/safe-stream/commit/4b9bb3266a537e59b10a2932f05dca47333420fc

**Comment by dg** [comment]
Great :+1:

**Pull request merged**